projects/cds/src/merchandising/adapters/strategy/cds-merchandising-strategy.adapter.spec.ts

```diff
@@ -125,7 +141,72 @@ describe('MerchandisingStrategyAdapter', () => {
       expect(cdsEndpointsService.getUrl).toHaveBeenCalledWith(
         STRATEGY_PRODUCTS_ENDPOINT_KEY,
         STRATEGY_ID_OBJECT,
-        STRATEGY_REQUEST
+        STRATEGY_REQUEST.queryParams
+      );
+
+      MOCK_STRATEGY_PRODUCTS_REQUEST.flush(STRATEGY_PRODUCTS);
+    });
+
+    it('should load the products for a given strategy id, the request URL should include the parameters in the StrategyRequest, and no HTTP header for consent-reference', () => {
+      strategyAdapter
+        .loadProductsForStrategy(
+          STRATEGY_ID,
+          STRATEGY_REQUEST_UNDEFINED_CONSENT_REFERENCE
+        )
+        .subscribe(strategyProducts => {
+          expect(strategyProducts).toEqual(STRATEGY_PRODUCTS);
```

Comment:
this wasn't asked before, but we can create a result variable with the specific type and set it here. Afterwards, we can unsubscribe and make the assertion right after like

.subscribe(strategyProducts => (result = strategyProducts))
.unsubscribe();

expect(result).toEqual(strategy_products)

Reply:
This doesn't work in tests because we are using a mock http server, which flushes the contents after the subscribe. So, if we have the code above, then the object isn't set, and the unsubscribe is cancelled before anything is emitted, causing the test to fail. I'm open to other ideas though :-)